packages/spark/sodasql/dialects/spark_dialect.py

```diff
@@ -114,7 +118,11 @@ def is_number(self, column_type: str):
             'FLOAT', 'DOUBLE', 'DOUBLE PRECISION', 'DECIMAL', 'NUMERIC']
 
     def qualify_table_name(self, table_name: str) -> str:
-        return f'{self.database}.{table_name}'
+        if self.database is None:
```

Comment:
Given the fact that `sql_tables_metadata` raises an Exception if database is None, we might never get here if both methods are used sequentially

Reply:
Correct. In `soda-spark` we will not run `sql_tables_metadata`, but use the `qualify_table_name` directly. And as mentioned when we implement #485, we can use the `schema` name here and allow to reference multiple schemas instead of one database (In spark databases are schemas, and vice versa)